src/main.cpp

```diff
@@ -117,64 +140,135 @@ void saveConfigCallback(){
+    ESP.rtcUserMemoryRead(0, (uint32_t*)&rtcData, sizeof(rtcData));
+    // Check if the magic number matches
+    if (rtcData.magicNumber != MAGIC_NUMBER) {
+      debugD("Invalid or uninitialized RTC memory.");
+    } else {
+      retVal = (rtcData.rebootFlag  == 1);
+    }
+  #elif defined(ESP32)
+    preferences.begin("reboot_data", false);
+    retVal = preferences.getBool("rebootFlag", false);
+    preferences.end();
+  #endif
+  rebootFlag = retVal;
+  return retVal;
+}
+
+void writeRebootFlag(bool flagValue) {
+  rebootFlag = flagValue;
+  #if defined(ESP8266)
+    // Write MAGIC_NUMBER so we can validate the data when we read it later
+    rtcData.magicNumber = MAGIC_NUMBER;
+    rtcData.rebootFlag = flagValue?1:0;
+    ESP.rtcUserMemoryWrite(0, (uint32_t*)&rtcData, sizeof(rtcData));
+  #elif defined(ESP32)
+    preferences.begin("reboot_data", false);
+    preferences.putBool("rebootFlag", flagValue);
+    preferences.end();
+  #endif
+}
+
+void startWiFiManager(){
+  writeRebootFlag(false);
+  if (ui.initialised) {
+    ui.server->stop();
+  }
+
+  WiFiManager wm;
+  WiFiManagerParameter custom_mqtt_server("server", "MQTT server", mqttServer.c_str(), 40);
+  WiFiManagerParameter custom_mqtt_port("port", "MQTT port", mqttPort.c_str(), 6);
+  WiFiManagerParameter custom_mqtt_username("username", "MQTT Username", mqttUserName.c_str(), 20 );
+  WiFiManagerParameter custom_mqtt_password("password", "MQTT Password", mqttPassword.c_str(), 40 );
+  wm.addParameter(&custom_mqtt_server);
+  wm.addParameter(&custom_mqtt_port);
+  wm.addParameter(&custom_mqtt_username);
+  wm.addParameter(&custom_mqtt_password);
+  wm.setBreakAfterConfig(true);
+  wm.setSaveConfigCallback(saveConfigCallback);
+
+
+  wm.startConfigPortal();
+  debugI("Exiting Portal");
+
+  if (saveConfig) {
+    mqttServer = String(custom_mqtt_server.getValue());
+    mqttPort = String(custom_mqtt_port.getValue());
+    mqttUserName = String(custom_mqtt_username.getValue());
+    mqttPassword = String(custom_mqtt_password.getValue());
+
+    debugI("Updating config file");
+    JsonDocument json;
+    json["mqtt_server"] = mqttServer;
+    json["mqtt_port"] = mqttPort;
+    json["mqtt_password"] = mqttPassword;
+    json["mqtt_username"] = mqttUserName;
+
+    File configFile = LittleFS.open("/config.json","w");
+    if (!configFile) {
+      debugE("Failed to open config file for writing");
+    } else {
+      serializeJson(json, configFile);
+      configFile.close();
+      debugI("Config file updated");
+    }
+  }
+}
+
+// We check the EN_PIN every loop, to allow people to configure the system
 void checkButton(){
 #if defined(EN_PIN)
   if(digitalRead(EN_PIN) == LOW) {
     debugI("Initial buttong press detected");
     delay(100); // wait and then test again to ensure that it is a held button not a press
     if(digitalRead(EN_PIN) == LOW) {
       debugI("Button press detected. Starting Portal");
-
-      if (ui.initialised) {
-        ui.server->stop();
```

Comment:
Are you sure that the WifiManager server does not conflict with the standard web server that we use to sever the web portal?

Reply:
See line 179. I moved all this code into a new function. Basically unchanged...